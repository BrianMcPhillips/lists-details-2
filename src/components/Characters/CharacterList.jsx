import React from 'react';
import PropTypes from 'prop-types';
import Character from './Character';

const CharacterList = ({ characters }) => {
  const characterStuff = characters.map(character => ( 
    <li key={character.name}>
      <Character {...character} />
    </li>
  ));

  return (
    <ul>
      {characterStuff}
    </ul>
  );
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    species: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired
};

export default CharacterList;
