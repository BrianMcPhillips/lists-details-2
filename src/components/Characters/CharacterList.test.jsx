import React from 'react';
import { render } from '@testing-library/react';
import CharacterList from './CharacterList';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterList component', () => {
  it('should render characters to page', () => {
    const characterStuff = [
      {
        name: 'Test 1',
        image: 'http://www.placecage.com/200/200'
      },
      {
        name: 'Test 2',
        image: 'http://www.placecage.com/300/300'
      }
    ];
    const { asFragment } = render(
      <MemoryRouter>
        <CharacterList characters={characterStuff}/>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
