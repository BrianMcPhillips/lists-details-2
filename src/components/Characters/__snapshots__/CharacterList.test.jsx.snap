// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component should render characters to page 1`] = `
<DocumentFragment>
  <ul
    data-testid="characters"
  >
    <li>
      <a
        href="/detail/undefined"
      >
        <h2>
          Test 1
        </h2>
        <img
          alt="Test 1"
          src="http://www.placecage.com/200/200"
        />
      </a>
    </li>
    <li>
      <a
        href="/detail/undefined"
      >
        <h2>
          Test 2
        </h2>
        <img
          alt="Test 2"
          src="http://www.placecage.com/300/300"
        />
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
